usaspending_api/etl/management/commands/reindex_elastic.py

```diff
@@ -0,0 +1,157 @@
+            type=str,
+            help='Prefix of output index')
+        parser.add_argument(
+            '--source_prefix',
+            default=settings.TRANSACTIONS_INDEX_ROOT,
+            type=str,
+            help='Prefix of index that has data')
+        parser.add_argument(
+            '--mapping_file',
+            type=str,
+            help='Absolute path to new mapping file')
+        parser.add_argument(
+            '--alias_prefix',
+            default='trans',
+            type=str,
+            help='Mapping file')
+
+    def handle(self, *args, **options):
+        ''' Script execution of custom code starts in this method'''
+        set_config()
+        self.source_prefix = options['source_prefix']
+        self.dest_prefix = options['dest_prefix']
+        self.alias_prefix = options['alias_prefix']
+        # TODO:
+        self.wipe = False
+        with open(options['mapping_file']) as f:
+            data = json.load(f)
+            self.mapping_file = json.dumps(data)
+
+        self.indicies_dict()
+        self.controller()
+
+    def single_reindex(self, source, dest):
+        logging.warn('calling create')
+        ES_CLIENT.indices.create(dest, body=self.mapping_file)
+        body_ = {
+                "source": {
+                        "index": source
+                        },
+                "dest": {
+                        "index": dest
+                        }
+                }
+        logging.warn('CREATING INDEX {}'.format(dest))
+        ES_CLIENT.reindex(body=body_)
+
+    def indicies_dict(self):
+        index_name = self.source_prefix + '*'
+        source_indices = ES_CLIENT.indices.get(index_name).keys()
+        source_indices = sorted(source_indices, key=lambda x: x.split('-')[-1])
+
+        dest_indices = [i.replace(self.source_prefix, self.dest_prefix) for i in source_indices]
+        self.reindex_dict = dict(zip(source_indices, dest_indices))
+        count = self.get_total_count(index_name)
+        logging.warn('Reindexing a total of {} documents'.format(count))
+
+    def get_total_count(self, index_name):
+        try:
+            response = ES_CLIENT.search(index=index_name)
+            return response["hits"]["total"]
+        except TransportError as e:
+            logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                          format(e=str(e)))
+            return -1
+
+    def verify(self, source, target):
+        difference = self.get_total_count(source) - self.get_total_count(target)
+        if difference != 0:
+            logging.warn('Still have {} documents to reindex'.format(difference))
+        return difference, difference/INGEST_RATE
+
+    def controller(self):
+        logging.warn('Reindexing the following \n')
+        pprint(self.reindex_dict)
+        print('\n\n')
+        for source, dest in self.reindex_dict.items():
+            if ES_CLIENT.indices.exists(dest):
+                if not self.wipe:
```

Comment:
rename to "delete_original" or "drop_original" to be more intuitive

Reply:
don't see `delete_original` in the script unless you mean the logging on line 116

Reply:
Sorry I wasn't more clear. `self.wipe` isn't intuitive. I recommend changing the variable name. The first comment has two suggestions